security/encrypt-cluster-pod-traffic.md

```diff
@@ -34,23 +34,107 @@ This how-to guide uses the following {{site.prodname}} features:
+To install WireGuard on the default Amazon Machine Image (AMI):
+
+   ```bash
+   sudo yum install kernel-devel-`uname -r` -y
+   sudo yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm -y
+   sudo curl -o /etc/yum.repos.d/jdoss-wireguard-epel-7.repo https://copr.fedorainfracloud.org/coprs/jdoss/wireguard/repo/epel-7/jdoss-wireguard-epel-7.repo
+   sudo yum install wireguard-dkms wireguard-tools -y
+   ```
+%>
+<label:OpenShift>
+<%
+To install WireGuard for OpenShift v4.3:
+
+   1. Create MachineConfig for WireGuard.
+   ```bash
+   cat <<EOF > mc-wg-worker.yaml
+   apiVersion: machineconfiguration.openshift.io/v1
+   kind: MachineConfig
+   metadata:
+     labels:
+       machineconfiguration.openshift.io/role: worker
+     name: 10-kvc-wireguard-kmod
+   spec:
+     config:
+   EOF
+   ```
+
+   2. Create base Ignition config.
+   ```bash
+   cat <<EOF > ./wg-config.ign
+   {
+     "ignition": { "version": "2.2.0" },
+     "systemd": {
+       "units": [{
+         "name": "require-kvc-wireguard-kmod.service",
+         "enabled": true,
+         "contents": "[Unit]\nRequires=kmods-via-containers@wireguard-kmod.service\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n\n[Install]\nWantedBy=multi-user.target"
+       }]
+     }
+   }
+   EOF
+   ```
+
+   3. Configure files.
+   ```bash
+   FAKEROOT=$(mktemp -d)
+   git clone https://github.com/kmods-via-containers/kmods-via-containers
+   cd kmods-via-containers
+   make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+   cd ..
+   git clone https://github.com/realgaurav/kvc-wireguard-kmod
+   cd kvc-wireguard-kmod
+   make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+   cd ..
+   ```
+
+   4. Configure RPMs for kernel-core, kernel-devel and kernel-modules for the host kernel (can be found by running uname -r on the host). Update `$FAKEROOT/etc/kvc/wireguard-kmod.conf` for the RPM location.
+
+   5. Get RHEL Entitlement data from your own RHEL8 system.
+   ```bash
+   [your-rhel8-host] # tar -czf subs.tar.gz /etc/pki/entitlement/ /etc/rhsm/ /etc/yum.repos.d/redhat.repo
+   ```
+
+   6. Copy the contents in the workspace and use the following command to add it to the MachineConfig.
+   ```bash
+   tar -x -C ${FAKEROOT} -f subs.tar.gz
+   ```
+
+   7. Get filetranspiler to generate the usable machine-config.
+   ```bash
+   git clone https://github.com/ashcrow/filetranspiler
+   ./filetranspiler/filetranspile -i ./wg-config.ign -f ${FAKEROOT} --format=yaml --dereference-symlinks | sed 's/^/     /' | (cat mc-wg-worker.yaml -) > mc-wg.yaml
+   ```
+
+   8. With with the KUBECONFIG set, run the following command to apply the MachineConfig created.
+   ```bash
+   oc create -f mc-wg.yaml
+   ```
+%>
+{% endtabs %}
+   > **Note**: Nodes that do not support WireGuard will not be secured by WireGuard tunnels, even if traffic running on the node to and from the pods goes to nodes that do support WireGuard.
    {: .alert .alert-info}
 
```

Comment:
#### Enable WireGuard

Use the following command to enable WireGuard encryption on all nodes.